### PR TITLE
feat: added examples of models and policies and added test for basic enforce function.

### PR DESCRIPTION
**Pull request by MrUnfunny**

### CODE REVIEW

**Comment by MrUnfunny** [comment]
@KNawm @santhoshivan23, I've added a test group for simple enforce. I've also left a commented out test group for the case when policyPath is empty. First test group is passing when the second group is commented out but it fails if the second group is also present. Somehow value of Enforcer in the first group is also changed to that of Enforcer in second group. I'm trying to look out for the reason for this and trying to replicate the issue in a smaller project to check if this is an issue for the testing package.

**Comment by hsluoyz** [comment]
@MrUnfunny fix:

image

**Comment by KNawm** [comment]
> @KNawm @santhoshivan23, I've added a test group for simple enforce. I've also left a commented out test group for the case when policyPath is empty. First test group is passing when the second group is commented out but it fails if the second group is also present. Somehow value of Enforcer in the first group is also changed to that of Enforcer in second group. I'm trying to look out for the reason for this and trying to replicate the issue in a smaller project to check if this is an issue for the testing package.

It may be because the Enforcer was first implemented as a singleton, I'm unable to test now but check the implementation of the Enforcer and make sure it doesn't store the instance in itself.

**Comment by santhoshivan23** [comment]
@MrUnfunny The Enforcer class is implemented as a singleton instance which means, only one Enforcer instance will be instantiated in memory. @KNawm, I think we followed this approach due to the factory constructor paradigm in Dart.

File: test/model/model_test.dart [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import 'package:test/test.dart';
+
+import 'package:casbin/src/enforcer.dart';
+
+import '../utils/test_utils.dart';
+
+void main() {
+  final modelPath = 'examples\\basic_model.conf';
+  final policyFile = 'examples\\basic_policy.csv';
```
Comment (santhoshivan23) [inline test/model/model_test.dart:9]: @MrUnfunny I had to manually change the file paths to the linux convention (forward slash) to run the tests.

Comment (santhoshivan23) [reply test/model/model_test.dart:9]: We should change this to something that makes it universally same for all platforms @MrUnfunny

Comment (MrUnfunny) [reply test/model/model_test.dart:9]: Hi @santhoshivan23, were you on Linux? For me, both '\\' and '/' works fine. I'll change it to '/'. It works on all systems. Right?

Comment (santhoshivan23) [reply test/model/model_test.dart:9]: yep!

**Comment by MrUnfunny** [comment]
@KNawm @santhoshivan23 I've removed the use of singleton in Enforcer. Please review the changes.

**Comment by santhoshivan23** [comment]
@MrUnfunny Just checked out. All test cases passed!

**Comment by santhoshivan23** [comment]
@hsluoyz PR can be landed.

**Review by KNawm — APPROVED** [review]

**Comment by KNawm** [comment]
We should get CI working ASAP as we now have working tests.
